File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const len=answers.length;
    const first = [1, 2, 3, 4, 5];
    const second = [2, 1, 2, 3, 2, 4, 2, 5];
    const third = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

    const scores = [0, 0, 0];  // 각각의 학생이 맞힌 문제 수를 저장

    for (let i = 0; i < len; i++) {
        if (answers[i] === first[i % first.length]) scores[0]++;
        if (answers[i] === second[i % second.length]) scores[1]++;
        if (answers[i] === third[i % third.length]) scores[2]++;
    }

    const maxScore = Math.max(...scores);  // 가장 높은 점수 찾기
    const result = [];

    for (let i = 0; i < 3; i++) {
        if (scores[i] === maxScore) {
            result.push(i + 1);  // 학생 번호는 1부터 시작하므로 i+1
        }
    }

    return result;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] p1=new int[]{1,2,3,4,5}; // 5개 반복
        int[] p2=new int[]{2,1,2,3,2,4,2,5}; // 8개 반복
        int[] p3=new int[]{3,3,1,1,2,2,4,4,5,5}; // 10개 반복
        int[] cnt=new int[3];
        
        for(int i=0; i<answers.length; i++){
            if(answers[i]==p1[i%5]) cnt[0]++;
            if(answers[i]==p2[i%8]) cnt[1]++;
            if(answers[i]==p3[i%10]) cnt[2]++;
        }
        List<Integer> answer = new ArrayList<>();
        int max = cnt[0];
        for(int i=1; i<=3; i++){
            int n=cnt[i-1];
            if(n>max){
                answer.clear();
                answer.add(i);
                max=n;
            }else if(n==max) answer.add(i);
        }

        int[] result=answer.stream().mapToInt(i->i).toArray();
        return result;
    }
}